### PR TITLE
Unit test examples with coroutines

### PR DESCRIPTION
**Pull request by agustinkoll-rootstrap**
#### ISSUE[#]
* Issue title

Unit test

#### Description
*

Somo unit test and TestDispatcherProvider to test coroutines
Added MockK lib to mock and verify suspend functions 
Injecting DispatcherProvider into VM's so can be mocked later

#### Risk
* No risk

### CODE REVIEW

File: app/src/test/java/com/rootstrap/android/SignInActivityViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
```
Comment (amaury901130) [inline app/src/test/java/com/rootstrap/android/SignInActivityViewModelTest.kt:25]: We must be aware of the changes for when it stops being experimental and update the version.

**Review by amaury901130 — APPROVED** [review]
Looking good, I'd be careful about the use of experimental features, we must be aware of the changes for when it stops being experimental and update the version.

**Review by marcossalto — APPROVED** [review]
Great work!

**Review by ximenaperez — APPROVED** [review]

**Comment by agustinkoll-rootstrap** [comment]
> Looking good, I'd be careful about the use of experimental features, we must be aware of the changes for when it stops being experimental and update the version.

Yes for now is the recommended way, but we need to be aware

**Pull request merged**